cli/sql.go

```diff
@@ -271,6 +272,15 @@ func runTerm(cmd *cobra.Command, args []string) error {
 	if len(cliContext.execStmts) > 0 {
 		// Single-line sql; run as simple as possible, without noise on stdout.
 		return runStatements(conn, cliContext.execStmts)
+	} else if !isatty.IsTerminal(os.Stdin.Fd()) {
+		// Non-tty stdin (eg: commands piped into cockroach): run quietly.
+		// TODO(marc): we may have to read through until ';' is encountered, but
```

Comment:
yup. and Tobi actually pointed out an example before I thought we had one. So this is a no-go.
The proper solution is to pass a reader and parse out individual (or blocks of) statements. Unfortunately, to do this properly, we probably need to go reach into the parser.